docroot/modules/humsci/hs_layouts/hs_layouts.module

```diff
@@ -113,3 +114,41 @@ function hs_layouts_preprocess_pattern_vertical_link_card(&$variables) {
     $variables['title_has_heading'] = _hs_layouts_field_has_title($field);
   }
 }
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function hs_layouts_form_entity_view_display_layout_builder_form_alter(&$form) {
+  $form['#validate'][] = '_hs_layouts_unique_main_content_section_validate';
+}
+
+/**
+ * Exactly one section should be set as 'Main Content' in Layout Builder form.
+ */
+function _hs_layouts_unique_main_content_section_validate(&$form, FormStateInterface $form_state) {
+  $main_content_found = FALSE;
+  foreach ($form['layout_builder']['#section_storage']->getSections() as $section) {
+    if ($section->getLayoutSettings()['main_content'] !== 'none') {
+      if ($main_content_found) {
+        $form_state->setErrorByName('form', 'Exactly one region should contain the main content id.');
+        break;
+      }
+      else {
```

Comment:
No need for an else block here if you return after setting the error.

Reply:
Fixed.